### ISSUE TITLE
DTLS

### ISSUE THREAD

**Issue by jvermillard** [telefonicaid/lwm2m-node-lib#86]
Is there an example on how to use it with DTLS (if supported?)
Thanks

**Comment by dmoranj** [comment]
Not really. The problem with the use of DTLS is that the underlying COAP library (node-coap) should support it before. There is an open discussion about that issue [here](https://github.com/mcollina/node-coap/issues/11). It references also some discussiones [here](https://github.com/nodejs/node-v0.x-archive/issues/25354), as it also depends on Node's features regarding DTLS. I would follow those threads and try to help if I have any spare time left.

**Comment by PedroMD** [comment]
There seems to be a fork of node-coap implementing dtls: https://github.com/neustar/node-coap-dtls

**Comment by caiothomas** [comment]
I implemented CoAP/DTLS in LWM2M/CoAP for university project.
In my Git in the [FIGUARDIAN](https://github.com/caiothomas?utf8=%E2%9C%93&tab=repositories&q=&type=fork&language=) branch has the security implementation in LWM2M/CoAP and SSL/TLS in the NGSI request in the IoT Agent.

**Comment by m4n3dw0lf** [comment]
In a partnership between the FiwareLab at Escola Politécnica da USP (Universidade de São Paulo) and FATEC SCS (Faculdade de Tecnologia de São Caetano do Sul), in Brazil, we figured out a way to implement a DTLS solution as described in this repository: https://github.com/m4n3dw0lf/SecureFiware with a PoC and documentations/guides about this security implementation and other security measures/analysis in the Fiware environment.

We modified the **lwm2m-node-lib** and the **lightweightm2m-iotagent** source code in 2 forks to support DTLS and to be simple to configure with the `.js` configuration file of the LWM2M IoTA, but we are now handling with npm-shrinkwrap issues and some refactoring to open the pull requests, the links of the forks:

- https://github.com/m4n3dw0lf/lwm2m-node-lib

- https://github.com/m4n3dw0lf/lightweightm2m-iotagent

The contacts can be found in the [SecureFiware](https://github.com/m4n3dw0lf/SecureFiware) repository, if any of you guys wants to discuss or propose improvements in the implementations.

**Comment by fgalan** [comment]
Maybe the best way to discuss implementation is to cast PRs in the respective repositories. That way, accurate in-line feedback can be provided.

CC: @dcalvoalonso

**Comment by reklatsmasters** [comment]
I`m working on purejs DTLS implementation https://github.com/nodertc/dtls. Ready for experiments and feedback.
